ambry-rest/src/main/java/com.github.ambry.rest/RestServer.java

```diff
@@ -157,8 +159,17 @@ public RestServer(VerifiableProperties verifiableProperties, ClusterMap clusterM
     RestRequestMetricsTracker.setDefaults(metricRegistry);
     restServerState = new RestServerState(restServerConfig.restServerHealthCheckUri);
     restServerMetrics = new RestServerMetrics(metricRegistry, restServerState);
+
+    boolean sslRequired = restServerConfig.restServerEnableHTTPS
+        || new ClusterMapConfig(verifiableProperties).clusterMapSslEnabledDatacenters.length() > 0;
+    SSLFactory sslFactory = null;
+    if (sslRequired) {
+      sslFactory = Utils.getObj(restServerConfig.restServerSSLFactory, verifiableProperties);
```

Comment:
Because there is no need for multiple implementations right now, I got rid of the extra interface, but it will be passed in from `RestServerMain`